src/test/java/io/jenkins/plugins/sshbuildagents/ssh/agents/ClientRSA512ConnectionTest.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright The Original Author or Authors
+ * SPDX-License-Identifier: MIT
+ */
+package io.jenkins.plugins.sshbuildagents.ssh.agents;
+
+import static hudson.plugins.sshslaves.tags.TestTags.AGENT_SSH_TEST;
+import static hudson.plugins.sshslaves.tags.TestTags.SSH_KEX_TEST;
+import static io.jenkins.plugins.sshbuildagents.ssh.agents.AgentConnectionBaseTest.SSH_PORT;
+import static io.jenkins.plugins.sshbuildagents.ssh.mina.ConnectionImpl.HEARTBEAT_INTERVAL;
+import static io.jenkins.plugins.sshbuildagents.ssh.mina.ConnectionImpl.HEARTBEAT_MAX_RETRY;
+import static io.jenkins.plugins.sshbuildagents.ssh.mina.ConnectionImpl.IDLE_SESSION_TIMEOUT;
+import static io.jenkins.plugins.sshbuildagents.ssh.mina.ConnectionImpl.WINDOW_SIZE;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.cloudbees.plugins.credentials.common.StandardUsernameCredentials;
+import io.jenkins.plugins.sshbuildagents.ssh.Connection;
+import io.jenkins.plugins.sshbuildagents.ssh.FakeSSHKeyCredential;
+import io.jenkins.plugins.sshbuildagents.ssh.ShellChannel;
+import io.jenkins.plugins.sshbuildagents.ssh.mina.ConnectionImpl;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.sshd.client.SshClient;
+import org.apache.sshd.client.channel.ChannelExec;
+import org.apache.sshd.client.channel.ClientChannelEvent;
+import org.apache.sshd.client.future.AuthFuture;
+import org.apache.sshd.client.future.ConnectFuture;
+import org.apache.sshd.client.session.ClientSession;
+import org.apache.sshd.common.util.io.input.NullInputStream;
+import org.apache.sshd.core.CoreModuleProperties;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+/**
+ * Connect to a remote SSH Server with a plain Apache Mina SSHD client.
+ *
+ */
+@Tag(AGENT_SSH_TEST)
+@Tag(SSH_KEX_TEST)
+@Testcontainers(disabledWithoutDocker = true)
+@DisabledOnOs(OS.WINDOWS)
```

Comment:
presumably because of the containers on windows in CI and windows containers and not Linux?

generally better to use a more specific exclusion (here and elsewhere).
e.g. (junit4) https://github.com/jenkinsci/ldap-plugin/pull/47